keps/sig-network/3178-iptables-cleanup/README.md

```diff
@@ -0,0 +1,839 @@
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Kubelet currently creates some IPTables chains at startup. In the past
+it actually used some of them, but with [the removal of dockershim] it
+no longer does. Additionally, for historical reasons, it creates some
+IPTables chains that are duplicates of chains also created by
+kube-proxy, and some that are only used by kube-proxy, but which
+kube-proxy requires kubelet to create on its behalf. (The initial
+comment in [kubernetes #82125] has more details on the history of how
+we got to where we are now, or at least to where we were before
+dockershim was removed.)
+
+We should clean this up so that kubelet no longer unnecessarily
+creates IPTables chains, and that kube-proxy creates all of the
+IPTables chains it needs.
+
+[the removal of dockershim]: ../../sig-node/2221-remove-dockershim
+[kubernetes #82125]: https://github.com/kubernetes/kubernetes/issues/82125
+
+## Motivation
+
+### Goals
+
+- Remove most IPTables chain management from kubelet.
+
+    - Remove the `--iptables-drop-bit` and `--iptables-masquerade-bit`
+      arguments from kubelet.
+
+    - Kubelet will continue to create at least one IPTables chain as a
+      hint to `iptables-wrapper` and other components that need to
+      know whether the system is using `legacy` or `nft` iptables.
+
+    - For security-backward-compatibility, kubelet will continue (for
+      now) to create a rule to block "martian packets" that would
+      otherwise be allowed by `route_localnet` (even though kubelet
+      itself does not set `route_localnet` or expect it to be set).
+      (See discussion in [Martian Packet
+      Blocking](#martian-packet-blocking) below.)
+
+- Update kube-proxy to no longer rely on chains created by kubelet:
+
+    - Rewrite packet-dropping rules to not depend on a
+      `KUBE-MARK-DROP` chain created by kubelet.
+
+    - Kube-proxy should create its own copy of kubelet's "martian
+      packet" fix, when running in a mode that needs that fix.
+
+- Ensure that the proxy implementations in `kpng` get updated
+  similarly.
+
+- Document for the future that Kubernetes's IPTables chains (other
+  than the IPTables mode "hint" chain) are meant for its internal use
+  only, do not constitute API, and should not be relied on by external
+  components.
+
+- Ensure that users and third-party software that make assumptions
+  about kubelet's and kube-proxy's IPTables rules have time to get
+  updated for the changes in rule creation.
+
+```
+<<[UNRESOLVED chain-renaming ]>>
+
+- (Maybe) Rename kube-proxy's `KUBE-MARK-MASQ` and `KUBE-POSTROUTING`
```

Comment:
At this point it seems pretty hostile.  Hyrum's law says this is part of our API now. :(